### PR TITLE
Change Ssl test to determine if results vary

### PR DESCRIPTION
**Pull request by steveharter**
Update this single test to determine impact; per https://github.com/dotnet/corefx/issues/16516

Will monitor for failures.

The IPAddress.Any to .Loopback change is subtle, assuming the test machines only have a single network card\interface. However, I believe .Any goes through a different code path as it doesn't short-circuit like using Loopback does.

The readBuffer was removed as it just throws an exception on background thread as connection is closed before it can finish.

Also change to close TcpListener.

cc @Priya91 @stephentoub

### CODE REVIEW

File: src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs [hunk 1]
```diff
@@ -16,12 +16,10 @@ namespace System.Net.Security.Tests
...
     public class SslStreamNetworkStreamTest
     {
-
         [Fact]
-        [ActiveIssue(16516, TestPlatforms.Windows)]
-        public async void SslStream_SendReceiveOverNetworkStream_Ok()
+        public async Task SslStream_SendReceiveOverNetworkStream_Ok()
         {
-            TcpListener listener = new TcpListener(IPAddress.Any, 0);
+            TcpListener listener = new TcpListener(IPAddress.Loopback, 0);
```
Comment (stephentoub) [inline src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs]: If I added this test, I probably just fat-fingered this with IntelliSense, as I would have intended Loopback in the first place, so good change :)

File: src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs [hunk 2]
```diff
@@ -62,9 +60,6 @@ public class SslStreamNetworkStreamTest
 
                     await Task.WhenAll(clientAuthenticationTask, serverAuthenticationTask);
 
-                    byte[] readBuffer = new byte[256];
-                    Task<int> readTask = clientStream.ReadAsync(readBuffer, 0, readBuffer.Length);
```
Comment (stephentoub) [inline src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs]: I'm not clear on why this is being removed.  The mistake here doesn't look like doing the read but rather doing the read but then not awaiting the resulting task... we should instead add the readTask to the WaitAll below.

Comment (steveharter) [inline src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs]: I wasn't sure of the intent. I will add in the resulting task to WaitAll

Comment (steveharter) [inline src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs]: Interesting it hangs when adding the readTask to WaitAll. Debugging...

Comment (stephentoub) [inline src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs]: It looks like it should be serverStream.ReadAsync, not clientStream.ReadAsync.

**Comment by ianhays** [comment]
Looks good Steve. Unfortunately, changing to loopback isn't applicable to the other tests failing in those way so I'd be interested to see if it resolves the failures for this one.

**Comment by steveharter** [comment]
Ian, it likely won't 'fix' this test but thought I would leave off the ActiveIssue for a while to see if there is any difference. This is a simple test that was flawed in more than one way, so I guess good to fix it anyway. Also, there is very small chance that returning void (instead of Task) may mess up SynchronizationContext depending on how XUnit runs these types of tests, which could lead to side affect problems.

**Pull request merged**